packages/editor/src/lib/TldrawEditor.tsx

```diff
@@ -191,11 +191,29 @@ export interface TldrawEditorBaseProps {
 	/**
 	 * Predicate for whether or not a shape should be hidden.
 	 *
+	 * @deprecated Use {@link TldrawEditorBaseProps#getShapeVisibility} instead.
+	 */
+	isShapeHidden?(shape: TLShape, editor: Editor): boolean
+
+	/**
+	 * Provides a way to hide shapes.
+	 *
 	 * Hidden shapes will not render in the editor, and they will not be eligible for hit test via
 	 * {@link Editor#getShapeAtPoint} and {@link Editor#getShapesAtPoint}. But otherwise they will
 	 * remain in the store and participate in all other operations.
+	 *
+	 * @example
+	 * ```ts
+	 * getShapeVisibility={(shape, editor) => shape.meta.hidden ? 'hidden' : 'inherit'}
+	 * ```
+	 *
+	 * @param shape - The shape to check.
+	 * @param editor - The editor instance.
 	 */
-	isShapeHidden?(shape: TLShape, editor: Editor): boolean
+	getShapeVisibility?(
+		shape: TLShape,
+		editor: Editor
+	): 'visible' | 'hidden' | 'inherit' | null | undefined | void
```

Comment:
Do we want `void` here? My understanding of it is that its meaning is closer to "we dont care about the return value of this function" than "nothing": https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgOrDACwGoHtgAmyA3gFACQCARgBQCUAXMgG74GkC+ppoksiKAHK48hEhWr0mAVxAEIMUBHZdSMWQjDBcIZJLrjyUCGGlRdAcjjIAzmCigA5hc7cEOu8gDuGHGyboWKJEALwkelTIXO4gniAi-sjCwchhxBFRQA